tests/publiccloud/nvidia.pm

```diff
@@ -20,11 +20,15 @@ sub run {
         zypper_call('--gpg-auto-import-keys addrepo -p 90 ' . get_required_var('NVIDIA_REPO') . ' nvidia_repo');
         zypper_call '--gpg-auto-import-keys ref';
     }
-    zypper_call("in nvidia-open-driver-G06-signed-kmp-default kernel-firmware-nvidia-gsp-G06 ", quiet => 1);
+    zypper_call("in nvidia-open-driver-G06-signed-kmp-default kernel-firmware-nvidia-gspx-G06 ", quiet => 1);
     $args->{my_instance}->softreboot(timeout => get_var('PUBLIC_CLOUD_REBOOT_TIMEOUT', 600));
 
     validate_script_output("hwinfo --gfxcard", sub { /nVidia.*Tesla T4/mg });    # depends on terraform setup
-    assert_script_run("LD_LIBRARY_PATH=/usr/lib/kernel-firmware-nvidia-gsp-G06 /usr/lib/kernel-firmware-nvidia-gsp-G06/nvidia-smi --query");
+        # assert_script_run("LD_LIBRARY_PATH=/usr/lib/kernel-firmware-nvidia-gsp-G06 /usr/lib/kernel-firmware-nvidia-gsp-G06/nvidia-smi --query");
+        # nvidia-smi is delivered by nvidia-compute-utils-G06. Not available on PublicCloud
+        # Check device files and loaded modules instead.
+    assert_script_run("ls -la /dev/{nvidia-modeset,nvidia-uvm-tools,nvidiactl,nvidia-uvm,nvidia0}");
+    assert_script_run("lsmod | grep nvidia");
```

Comment:
I would do this before checking the device files. If the kernel module is not loaded then the device files are not there. Also WDYT about

```suggestion
    validate_script_output("lsmod", sub { m/nvidia/ }, fail_message => "nvidia module not loaded");
```

Reply:
*ready for review*

Reply:
Please move this check up. If the module is not loaded, the device files are not present, so this is basically dead code. If you move it up, it acts as a preliminary check if the kernel modules is loaded before the device files are checked.

Reply:
@grisu48 please check again. change was pushed

Reply:
yep, looks good now. Thanks!